ebpf/xdp_filter.c

```diff
@@ -388,6 +398,16 @@ static int __always_inline filter_ipv6(void *data, __u64 nh_off, void *data_end,
         return XDP_PASS;
     }
 #else
+#if RSS_QUEUE_NUMBERS
+    /* IP-pairs + protocol (UDP/TCP/ICMP) hit same CPU */
+    __u32 xdp_hash  = tuple.src[0] + tuple.dst[0];
+    xdp_hash += tuple.src[1] + tuple.dst[1];
+    xdp_hash += tuple.src[2] + tuple.dst[2];
+    xdp_hash += tuple.src[3] + tuple.dst[3];
+    xdp_hash = SuperFastHash((char *)&xdp_hash, 4, ip6h->nexthdr);
```

Comment:
this seems like a bad idea: nexthdr is not the same as ipproto. It can be a lot of values for various ip6 opts. Many of them can be added between the ip6 hdr and the L4 protocol. This would need to resolve to the same hash value.

So either you need to walk all the ip6 options to get to the ipproto, or you need to not use the ipproto at all. Same applies to the BUILD_CPUMAP block above.